code/game/objects/items/tools/welder.dm

```diff
@@ -122,8 +122,8 @@
 /obj/item/weldingtool/tool_check_callback(mob/living/user, amount, datum/callback/extra_checks)
 	. = ..()
 	if(. && user)
-		if(progress_flash_divisor == 0)
-			user.flash_eyes(min(light_intensity, 1))
+		if(progress_flash_divisor <= 0)
+			user.add_flash_overlay(min(light_intensity, 1))
```

Comment:
flash_eyes checked for eye protection before doing the flash - this doesn't

Reply:
He moved the protection check to add_flash_overlay and made flash_eyes call add_flash_overlay as a base.
So the protection check will be done but only the eye damage and other stuff flash_eyes might do will not be applied